KEMField/Source/Interface/Fields/Magnetic/include/KMagneticSuperpositionField.hh

```diff
@@ -55,6 +67,9 @@ class KMagneticSuperpositionField : public KMagneticField
 
     bool fUseCaching;
     bool fCachingBlock;
+    bool fRequireAll;
```

Comment:
I would suggest using an `enum` for this instead of three mutually exclusive fields. This should also make the code more readable.